bazel/thirdparty/openssl-fips.BUILD

```diff
@@ -1,18 +1,33 @@
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
 load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make")
 
-filegroup(
-    name = "srcs",
-    srcs = glob(["**"]),
+string_flag(
+    name = "build_mode",
+    build_setting_default = "default",
+    values = [
+        "debug",
+        "release",
+        "default",
+    ],
 )
```

Comment:
Should this and the non-fips flags always be the same? Should we consildate and have the config_settings in this file reference `@openssl//:build_mode`? You'll have to set the visibility

Reply:
yeh, definitely, fixed. i didn't even try before because i was getting burned in the recent past by visibility rules that come with bazel modules. but it seems to work as-is without additional changes.

Reply:
yeh same. i verified that it seems to be doing what we expect by looking at the generated foregn_cc build scripts.